src/ui/molecules/stripe-payment-elements.svelte

```diff
@@ -232,82 +235,117 @@
-            }
-
-            onPaymentInfoChange({
-              complete: event.complete,
-              paymentMethod: event.complete ? event.value.type : undefined,
-            });
-          },
-        );
-        paymentElement.on("loaderror", (event) => {
-          isMounted = false;
-          onError({
-            code: PaymentElementErrorCode.ErrorLoadingStripe,
-            gatewayErrorCode: event.error.code,
-            message: event.error.message,
-          });
-          onLoadingComplete();
-        });
-      } catch (error) {
-        if (!isMounted) return;
+      paymentElement = StripeService.createPaymentElement(
+        unsafeElements,
+        brandingInfo?.app_name,
+      );
 
+      paymentElement.mount("#payment-element");
+
+      paymentElement.on("ready", () => {
+        onLoadingComplete();
+      });
+
+      paymentElement.on(
+        "change",
+        async (event: StripePaymentElementChangeEvent) => {
+          if (
+            taxCollectionEnabled &&
+            event.complete &&
+            event.value.type === "card"
+          ) {
+            await triggerTaxDetailsUpdated();
+          }
+
+          onPaymentInfoChange({
+            complete: event.complete,
+            paymentMethod: event.complete ? event.value.type : undefined,
+          });
+        },
+      );
+      paymentElement.on("loaderror", (event) => {
+        isMounted = false;
         onError({
           code: PaymentElementErrorCode.ErrorLoadingStripe,
-          gatewayErrorCode: undefined,
-          message: error instanceof Error ? error.message : String(error),
+          gatewayErrorCode: event.error.code,
+          message: event.error.message,
         });
         onLoadingComplete();
-      }
-    })();
+      });
+    } catch (error) {
+      if (!isMounted) return;
+
+      onError({
+        code: PaymentElementErrorCode.ErrorLoadingStripe,
+        gatewayErrorCode: undefined,
+        message: error instanceof Error ? error.message : String(error),
+      });
+      onLoadingComplete();
+    }
+  };
 
+  onMount(() => {
+    isMounted = true;
+    mountStripeElements();
     return () => {
       if (isMounted) {
         isMounted = false;
         paymentElement?.destroy();
+        addressElement?.destroy();
+        addressElement = null;
       }
     };
   });
+
+  $: (() => {
+    if (!isMounted || !elements || !paymentElement) {
+      return;
+    }
+
+    if (priceBreakdown.taxCalculationBasedOnFullAddress) {
+      if (addressElement !== null) {
+        return;
+      }
+      addressElement = StripeService.createAddressElement(elements);
+      addressElement.mount("#address-element");
+      addressElement.on(
+        "change",
+        async (event: StripeAddressElementChangeEvent) => {
+          if (taxCollectionEnabled && event.complete && event.isNewAddress) {
+            await triggerTaxDetailsUpdated();
+          }
+        },
+      );
```

Comment:
Would it make sense to move the `initialize` and `destroy` logic for this element into `onMount()`? Then here, we’d just call `addressElement.mount("#address-element")` and `addressElement.unmount()` as needed. Curious if that might help simplify things a bit.

Reply:
What I mean is, initializing the address element only once on mount and call only mount/unmount (add the element to the DOM) at this point.

Reply:
I looks like we need to create this element on demand to set [defaultValues](https://docs.stripe.com/js/elements_object/create_address_element#address_element_create-options-defaultValues). So lets keep your initial approach

Reply:
I am refactoring the implementation that got too complex.

We'll have something like

```
{#if elements}
 <PaymentElement {...}/>
 {#if priceBreakdown.taxCalculationBasedOnFullAddress}
 <AddressElement {...}/>
 {/if}
{/if}


```